src/log_database_proxy_model.cpp

```diff
@@ -244,6 +251,83 @@ bool LogDatabaseProxyModel::isExcludeValid() const
   return true;
 }
 
+//Locates the next index based on search criteria, VCM 25 April 2017
+//searchText_ - string from searchText, all upper case and trimmed spaces
+//index - currently selected item in messageList
+//increment - +1 = next||search(i.e. down), -1 = prev (i.e. up)
+int LogDatabaseProxyModel::getItemIndex(const QString searchText_, int index, int increment)
+{
+	int searchNotFound = -1;//indicates search not found
+	int counter=0;//used to stop loop once full list has been searched
+	bool partialSearch = false;//tells main loop to run a partial search, triggered by prior failed search
+	if(searchText_==""||msg_mapping_.size()==0)//skip search for 1)empty string 2)empty set
+	{
+		clearSearchFailure(); //reset failed search variables
+		return searchNotFound;
+	}
+
+	//round corners for searches
+	if(index<0)//if index < 0, set to size()-1;
+	{
+		index = msg_mapping_.size()-1;
+	}
+	else if(index>=msg_mapping_.size())//if index >size(), set to 0;
+	{
+		index = 0;
+	}
+
+//trigger partial search if:
+	//searchText_ conaints prior failed text
+	//prior failed text is not empty
+	//failed index is not 0
+	//failed search index isn't greater than current index, this could happen through user interface message selection. Software should clear the variables when UI is adjusted.
+	if(searchText_.contains(failedSearchText) && failedSearchText != "" && failedSearchIndex !=0 && failedSearchIndex <= msg_mapping_.size() )
+	{
+		partialSearch = true;
+		index = failedSearchIndex-1;
+		counter = failedSearchIndex;
+	}
+
+	while(true)//loop through all messages until end or match is found
```

Comment:
Infinite `while` loops that depend on conditional code inside them to break out make me nervous.  Any reason not to make this a `for` loop that iterates over `msg_mapping_`?